jovo-framework/src/middleware/Handler.ts

```diff
@@ -334,9 +334,51 @@ export class Handler implements Plugin {
         Jovo.prototype.toStateIntent = async function (
             state: string | undefined,
             intent: string,
+        ): Promise<any> {
+            // Check for Component Names inside State
+            if (typeof state !== 'undefined' && this.$components) {
```

Comment:
Yes moving the 

> instead of using `if (typeof state !== 'undefined' && this.$components)` use ` if (state && this.$components)` please.
> 
> Keeps it consistent with the rest of the codebase.

I had that prior to the typeof stuff, but strangely I got some tslint errors with that, but I will try to change it back to simpler type checking. 

> I think we should move that part into its own function. That way we reduce some redundancy.

Yes, good point!

> My first idea as for the name of it would be `isStateSameAsIntializedComponentName()`. But since that's fairly long, I'm open for other alternatives.

There's nothing wrong with a long name, as **long** (pardon the pun) it's easier to understand. `checkStateForInitializedComponentName` would also fit in my option, also if you like it shorter we might just strip the "Initialized".

Reply:
I like the name. Let's go with that.